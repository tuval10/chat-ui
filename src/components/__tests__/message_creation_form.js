import React from 'react';
import MessageCreationForm from '../message_creation_form';
import {noop} from 'lodash';

import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

const actions = {
  validate: noop,
  onSubmit: noop,
  handleSubmit: func => func
};

const createShallowForm = (props) =>
  shallow(<MessageCreationForm {...actions} {...props}/>);

describe('MessageItem component', () => {
  describe('renders correctly', () => {
    it('when form is pristine', () => {
      expect(createShallowForm({pristine: true, submitting: false})).toMatchSnapshot();
    });

    it('when submitting', () => {
      expect(createShallowForm({pristine: false, submitting: true})).toMatchSnapshot();
    });

    it('when ready to submit', () => {
      expect(createShallowForm({pristine: false, submitting: false})).toMatchSnapshot();
    });
  });
});