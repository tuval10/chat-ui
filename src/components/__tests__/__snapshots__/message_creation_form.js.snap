// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem component renders correctly when form is pristine 1`] = `
<form
  className="form-container"
  onSubmit={[Function]}
>
  <div
    className="fields-container"
  >
    <Field
      component={[Function]}
      name="username"
      placeholder="Your nickname..."
      type="text"
    />
    <Field
      component={[Function]}
      name="text"
      placeholder="Add a message..."
      type="text"
    />
  </div>
  <div
    className="send-container"
  >
    <button
      className="send-button brand-bg-color"
      disabled={true}
      type="submit"
    >
      Send
    </button>
  </div>
</form>
`;

exports[`MessageItem component renders correctly when ready to submit 1`] = `
<form
  className="form-container"
  onSubmit={[Function]}
>
  <div
    className="fields-container"
  >
    <Field
      component={[Function]}
      name="username"
      placeholder="Your nickname..."
      type="text"
    />
    <Field
      component={[Function]}
      name="text"
      placeholder="Add a message..."
      type="text"
    />
  </div>
  <div
    className="send-container"
  >
    <button
      className="send-button brand-bg-color"
      disabled={false}
      type="submit"
    >
      Send
    </button>
  </div>
</form>
`;

exports[`MessageItem component renders correctly when submitting 1`] = `
<form
  className="form-container"
  onSubmit={[Function]}
>
  <div
    className="fields-container"
  >
    <Field
      component={[Function]}
      name="username"
      placeholder="Your nickname..."
      type="text"
    />
    <Field
      component={[Function]}
      name="text"
      placeholder="Add a message..."
      type="text"
    />
  </div>
  <div
    className="send-container"
  >
    <button
      className="send-button brand-bg-color"
      disabled={true}
      type="submit"
    >
      Send
    </button>
  </div>
</form>
`;
